---
title: "C6320 | Microsoft Docs"
ms.custom: ""
ms.date: "12/03/2016"
ms.prod: "visual-studio-dev14"
ms.reviewer: ""
ms.suite: ""
ms.technology: 
  - "vs-devops-test"
ms.tgt_pltfrm: ""
ms.topic: "article"
f1_keywords: 
  - "C6320"
helpviewer_keywords: 
  - "C6320"
ms.assetid: fb9e568e-b3d4-4ce2-a276-a64ad74d7b1e
caps.latest.revision: 10
caps.handback.revision: 10
author: "corob-msft"
ms.author: "corob"
manager: "ghogen"
---
# C6320
[!INCLUDE[vs2017banner](../code-quality/includes/vs2017banner.md)]

aviso C6320: A expressão do filtro de exceção é a constante EXCEPTION\_EXECUTE\_HANDLER.  Isso pode mascarar exceções que não foram feitas para serem tratadas  
  
 Esse aviso indicar o efeito colateral de usar a constante de EXCEPTION\_EXECUTE\_HANDLER no bloco de \_\_except.  Nesse caso, a instrução no bloco de \_\_except será executado sempre para tratar a exceção, inclusive exceções que você não quiser tratar em uma função específica.  É recomendado que você verifique a exceção antes de ser identificado.  
  
## Exemplo  
 O código a seguir gerenciar esse aviso como o bloco de \_\_except tentará tratar exceções de todos os tipos:  
  
```  
#include <stdio.h>   
#include <excpt.h>   
  
void f(int *p)   
{   
   __try  
   {   
      puts("in try");   
      *p = 13;  // might cause access violation exception  
      // code ...  
   }   
   __except(EXCEPTION_EXECUTE_HANDLER) // warning  
   {   
      puts("in except");   
      // code ...  
   }   
}   
```  
  
 Para corrigir esse aviso, use `GetExceptionCode` para verificar uma exceção específico antes que executando a conforme mostrado no seguinte código:  
  
```  
#include <stdio.h>   
#include <windows.h>   
#include <excpt.h>   
  
void f(int *p)   
{   
   __try  
   {   
      puts("in try");   
      *p = 13;    // might cause access violation exception   
      // code ...  
   }   
   __except(GetExceptionCode()==EXCEPTION_ACCESS_VIOLATION ?   
               EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH)  
   {   
      puts("in except");   
      // code ...  
   }   
}  
```  
  
## Consulte também  
 [Instrução try\-except](/visual-cpp/cpp/try-except-statement)